eodatasets/prepare/ncep_reanalysis_surface_pr_wtr.py

```diff
@@ -0,0 +1,166 @@
+
+
+NOAA_WATER_VAPOUR_NS = uuid.UUID(hex='857bd048-8c86-4670-a2b4-5dbea26d7692')
+NOAA_DESCRIPTION = ''
+
+
+def get_coords(collection, xkey='lon', ykey='lat'):
+    """
+    transforms the bounds of a collection into an
+    encompassing bounding box. Keys for x and y are
+    customisable
+    """
+    return {
+        'll': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.left
+        },
+        'lr': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.right
+        },
+        'ul': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.left
+        },
+        'ur': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.right
+        }
+    }
+
+
+def get_uuid(collection, idx):
+    """
+    Returns a deterministic uuid based on band index and gdal checksum
+    """
+    origin = collection.tags()['NC_GLOBAL#References']
+
+    return uuid.uuid5(
+        NOAA_WATER_VAPOUR_NS,
+        '{}?{}'.format(origin, urllib.parse.urlencode({
+            'checksum': collection.checksum(idx),
+            'band_index': idx
+        }))
+    )
+
+
+def process_datasets(dataset: Path):
+    """
+    Generates a metadata document for each band available
+    in the water vapour file.
+
+    Each band is treated as a separate dataset since the
+    source file is updated in-place; allowing the file
+    to be modified but a static reference is provided for
+    each observation
+
+    """
+    datasets = []
+    creation_dt = (
+        datetime.datetime.utcnow()
+        .replace(tzinfo=datetime.timezone.utc)
+    )
+
+    with rasterio.open(str(dataset), 'r') as collection:
+        collection_start_date = (
+            datetime.datetime.strptime(
+                collection.tags()['time#units'],
+                'hours since %Y-%m-%d %H:%M:%S.%f'
+            )
+        )
+        creation_dt = (
+            datetime.datetime.utcnow()
+            .replace(tzinfo=datetime.timezone.utc)
+        )
+
+        for _idx in collection.indexes:
+            time_in_hours = int(collection.tags(_idx)['NETCDF_DIM_time'])
+            ds_dt = (
+                collection_start_date +
+                datetime.timedelta(hours=time_in_hours)
+            ).replace(tzinfo=datetime.timezone.utc)
+
+            md = {}
+            md['id'] = get_uuid(collection, _idx).hex
+            md['creation_dt'] = creation_dt.isoformat()
+            md['description'] = NOAA_DESCRIPTION
+            md['extent'] = {
+                'center_dt': ds_dt.isoformat(),
+                'coord': get_coords(collection)
+            }
+            md['grid_spatial'] = {
+                'projection': {
+                    'geo_ref_points': get_coords(collection, xkey='x', ykey='y'),
+                    'spatial_reference': rasterio.crs.CRS(init='EPSG:4236').wkt
+                }
+            }
+            md['format'] = {'name': 'netCDF'}
```

Comment:
Most of the dea products use capitalisation "NetCDF" from the looks of it: https://github.com/GeoscienceAustralia/digitalearthau/search?q=netcdf&unscoped_q=netcdf

(though humorously there's deviations in other repositories :smile: 
![netcdf-variations](https://user-images.githubusercontent.com/25688/56264931-73744b00-612b-11e9-8877-80f2cc835119.png))